### PR TITLE
[ML] all multiple wildcard values for GET Calendars, Events, and DELETE forecasts

### PR DESCRIPTION
**Pull request by benwtrent**
This commit adjusts the following APIs so now they not only support an `_all` case, but wildcard patterned Ids as well.

- `GET _ml/calendars/<calendar_id>/events`
- `GET _ml/calendars/<calendar_id>`
- `GET _ml/anomaly_detectors/<job_id>/model_snapshots/<snapshot_id>`
- `DELETE _ml/anomaly_detectors/<job_id>/_forecast/<forecast_id>`

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/QueryBuilderHelper.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Optional;
+
+public final class QueryBuilderHelper {
+
+    private QueryBuilderHelper() { }
+
+    /**
+     * Helper function for adding OR type queries for a given identity field.
+     *
+     * This adds a new filter to the passed `boolQueryBuilder`.
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/QueryBuilderHelper.java:27]:
```suggestion

```

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/ResultsFilterBuilder.java [hunk 2]
```diff
@@ -88,6 +89,11 @@ public ResultsFilterBuilder term(String fieldName, String fieldValue) {
         return this;
     }
 
+    public ResultsFilterBuilder resourceTokenFilers(String fieldName, String[] tokens) {
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/ResultsFilterBuilder.java:92]:
```suggestion
    public ResultsFilterBuilder resourceTokenFilters(String fieldName, String[] tokens) {
```

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetCalendarsAction.java [hunk 3]
```diff
@@ -34,35 +31,18 @@ public TransportGetCalendarsAction(TransportService transportService, ActionFilt
...
-        final String calendarId = request.getCalendarId();
-        if (request.getCalendarId() != null && Strings.isAllOrWildcard(request.getCalendarId()) == false) {
-            getCalendar(calendarId, listener);
-        } else {
-            PageParams pageParams = request.getPageParams();
-            if (pageParams == null) {
-                pageParams = PageParams.defaultParams();
-            }
-            getCalendars(pageParams, listener);
+        final String[] calendarIds = Strings.splitStringByCommaToArray(request.getCalendarId());
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetCalendarsAction.java:34]: The GET request docs will need updating to add the standard description of a wildcard-able ID parameter

"comma-separated list of calendar Ids and wildcard expressions. You can get all by using `_all`, or `*` or by omitting the ID"

**Review by davidkyle — APPROVED** [review]
LGTM

Thanks for all the tests

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**